src/tools/jsondocck/src/main.rs

```diff
@@ -220,6 +234,41 @@ fn check_command(command: Command, cache: &mut Cache) -> Result<(), CkError> {
                 Err(_) => false,
             }
         }
+        CommandKind::Is => {
+            // @has <path> <jsonpath> <value> = check *exactly one* item matched by path, and it equals value
+            assert_eq!(command.args.len(), 3);
+            let val = cache.get_value(&command.args[0])?;
+            match select(&val, &command.args[1]) {
+                Ok(results) => {
+                    let v_holder;
+                    let pat: &Value = if command.args[2].starts_with("$") {
+                        &cache.variables[&command.args[2][1..]]
+                    } else {
+                        v_holder = serde_json::from_str(&command.args[2]).unwrap();
+                        &v_holder
+                    };
+                    results.len() == 1 && results[0] == pat
+                }
+                Err(_) => false,
+            }
+        }
+        // FIXME, Figure out semantics for @!set
```

Comment:
What is this fixme for?

Reply:
Currently `@!set` will check that the the `select` is `Err`, but this is just `@!has` which is much clearer.

is not, has not, and count not all have clear meanings, but not set only makes sense if you know how jsondocck works.

The answer is problably to pass weater we are `!` or not to `check_command`, to allow better error reporting, and remove `@!set`, and probably `@!count` and `@!is`, but not now

Reply:
Hmm, rather than giving an error, could you panic instead and say that the path wasn't parsed? I wouldn't expect *either* `@set` or `@!set` to be able to fail.

Reply:
Done in 4c949a455df81924b37dc41f2edf1071ea329fb5, `@set` will fail if the selector doesnt return exactly one item, `@!set` will fail if the selector matches something, both will fail if multiple items are selected, which I think is the most sensable semantics.